template/go/Dockerfile

```diff
@@ -1,32 +1,41 @@
 FROM openfaas/classic-watchdog:0.18.1 as watchdog
-FROM golang:1.12-alpine3.11 as builder
+FROM golang:1.13-alpine3.11 as builder
 
 # Allows you to add additional packages via build-arg
 ARG ADDITIONAL_PACKAGE
 ARG CGO_ENABLED=0
 ARG GO111MODULE="off"
 ARG GOPROXY=""
+ARG GOFLAGS=""
 
 COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
 RUN chmod +x /usr/bin/fwatchdog
 
+ENV CGO_ENABLED=0
+
 WORKDIR /go/src/handler
 COPY . .
 
+# Add user overrides to the root go.mod, which is the only place "replace" can be used
+RUN cat function/GO_REPLACE.txt >> ./go.mod || exit 0
```

Comment:
I find the GO_REPLACE pattern a bit confusing, have you consider merging `go.mod` instead? That way the function writer still has a regular `go.mod` that can be linted / validated. This project seems like it could do the trick: https://github.com/brendanjryan/modmerge

Reply:
Or maybe even go the multi-module approach and have the `main` code can be a module and the function code a different one? If a package name is enforced we might not even need the replace.  https://github.com/golang/go/wiki/Modules#faqs--multi-module-repositories has an overview of that approach.

Reply:
This multi-module approach is what we are doing here, but ... go does not follow replaces in modules, so if you need to use a replace in your function handler module, it will not be respected in the parent template module. This GO_REPLACE is a way for us to explicitly allow overriding the replaces in the parent function module.

Reply:
https://github.com/openfaas/templates/pull/194#issuecomment-580947149